android/java/org/chromium/chrome/browser/upgrade/BraveUpgradeJobIntentService.java

```diff
@@ -53,12 +54,51 @@
     private static final String DSE_NAME = "Google";
     private static final String DSE_KEYWORD = "google.com";
 
+    // Old tabs bottom toolbar settings
+    private static final String BOTTOM_TOOLBAR_ENABLED_KEY = "bottom_toolbar_enabled";
+
+    // To detect update from tabs
+    private static final String PREF_STATS_PREFERENCES_NAME = "StatsPreferences";
+    private static final String PREF_WEEK_OF_INSTALLATION_NAME = "WeekOfInstallation";
+
     public static void startMigrationIfNecessary(Context context) {
+        if (BraveUpgradeJobIntentService.needToMigratePreferences()) {
+            // Migrate bottom toolbar settings
+            SharedPreferences sharedPreferences = ContextUtils.getAppSharedPreferences();
+            SharedPreferences.Editor sharedPreferencesEditor = sharedPreferences.edit();
+            sharedPreferencesEditor.putBoolean(BravePreferenceKeys.BRAVE_BOTTOM_TOOLBAR_ENABLED_KEY,
+                sharedPreferences.getBoolean(BOTTOM_TOOLBAR_ENABLED_KEY, true));
+            sharedPreferencesEditor.apply();
+        }
```

Comment:
Have to call it before the scheduled job as it allows to run non heave tasks. Otherwise we don't have the settings migrated on a first task sometimes.